nxc/modules/pre2k.py

```diff
@@ -0,0 +1,150 @@
+from impacket.krb5.ccache import CCache
+from impacket.krb5.types import Principal
+from impacket.krb5 import constants
+
+class NXCModule:
+    """
+    Identify pre-created computer accounts, save the results to a file, and obtain TGTs for each pre-created computer account.
+    Module by : @shad0wcntr0ller
+    
+    """
+    name = "pre2k"
+    description = "Identify pre-created computer accounts, save the results to a file, and obtain TGTs for each"
+    supported_protocols = ["ldap"]
+    opsec_safe = True
+    multiple_hosts = False
+
+    def options(self, context, module_options):
+        pass
+
+    def on_login(self, context, connection):
+        try:
+            # Initialize connection to LDAP
+            context.log.info(f"Connecting to LDAP server at ldap://{connection.host}")
+
+            if connection.kerberos:
+                ldap_connection = ldap.LDAPConnection(f"ldap://{connection.host}", connection.baseDN, None)
+                ldap_connection.kerberosLogin(connection.username, connection.password, connection.domain, lmhash=connection.lmhash, nthash=connection.nthash, aesKey=connection.aesKey, kdcHost=connection.kdcHost)
+            else:
+                ldap_connection = ldap.LDAPConnection(f"ldap://{connection.host}", connection.baseDN, None)
+                ldap_connection.login(connection.username, connection.password, connection.domain, lmhash=connection.lmhash, nthash=connection.nthash)
+
+            # Define the search filter for pre-created computer accounts
+            search_filter = "(&(objectClass=computer)(userAccountControl=4128))"
+            attributes = ["sAMAccountName", "userAccountControl", "dNSHostName"]
+
+            context.log.info(f"Using search filter: {search_filter}")
+            context.log.info(f"Attributes to retrieve: {attributes}")
+
+            computers = []
+
+            try:
+                # Use paged search to retrieve all computer accounts with specific flags
+                paged_search_control = ldapasn1.SimplePagedResultsControl(criticality=True, size=1000)
+                search_results = ldap_connection.search(searchFilter=search_filter, attributes=attributes, searchControls=[paged_search_control])
+
+                for item in search_results:
+                    if isinstance(item, ldapasn1.SearchResultEntry):
+                        context.log.debug(f"Raw item: {item.prettyPrint()}")
+
+                        sam_account_name = None
+                        user_account_control = None
+
+                        for attribute in item["attributes"]:
+                            context.log.debug(f"Attribute: {attribute.prettyPrint()}")
+                            if str(attribute["type"]) == "sAMAccountName":
+                                sam_account_name = str(attribute["vals"][0])
+                            elif str(attribute["type"]) == "userAccountControl":
+                                user_account_control = str(attribute["vals"][0])
```

Comment:
This could be replaced by our internal searching and parsing functionality. With `self.search(searchFilter, attributes)` you can query and parse the output with `parse_result_attributes(response)` (take a look at the `def users` function in ldap.py). This will return a beautifully parsed response object 

Ref `search`:
https://github.com/Pennyw0rth/NetExec/blob/3a5c109fedbcccda8895f134dfa9d078fc0cd8b2/nxc/protocols/ldap.py#L697
Ref `users`:
https://github.com/Pennyw0rth/NetExec/blob/3a5c109fedbcccda8895f134dfa9d078fc0cd8b2/nxc/protocols/ldap.py#L720

Reply:
Fixed